app/Filament/Pages/Installer/Steps/SessionStep.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Filament\Pages\Installer\Steps;
+
+use Filament\Forms\Components\TextInput;
+use Filament\Forms\Components\ToggleButtons;
+use Filament\Forms\Components\Wizard\Step;
+use Filament\Forms\Get;
+
+class SessionStep
+{
+    public const SESSION_DRIVERS = [
+        'file' => 'Filesystem',
+        'database' => 'Database',
+        'cookie' => 'Cookie',
+        'redis' => 'Redis',
+    ];
+
+    public static function make(): Step
+    {
+        return Step::make('session')
+            ->label('Session')
+            ->schema([
+                ToggleButtons::make('env_session.SESSION_DRIVER')
+                    ->label('Session Driver')
+                    ->hintIcon('tabler-question-mark')
+                    ->hintIconTooltip('The driver used for storing sessions. We recommend "Filesystem" or "Database".')
+                    ->required()
+                    ->inline()
+                    ->options(self::SESSION_DRIVERS)
+                    ->disableOptionWhen(fn ($value, Get $get) => $value === 'redis' && $get('env_cache.CACHE_STORE') !== 'redis')
+                    ->default(config('session.driver')),
+                TextInput::make('env_session.SESSION_SECURE_COOKIE')
+                    ->hidden()
+                    ->default(str_starts_with(url(''), 'https://') ? 'true' : 'false'),
```

Comment:
```suggestion
                    ->default(request()->isSecure()),
```

Reply:
`request()->isSecure()` might not always return the correct value, e.g. when using proxies.
Checking the url is not the prettiest but it works.

Reply:
https://github.com/pelican-dev/panel/blob/cb522b24ef79f372454a55042fcb5668e322482c/app/Providers/AppServiceProvider.php#L44-L46 should handle that just fine and we use `request()->isSecure()` everywhere else i don't see why wouldn't it work here.